python/ray/_private/client_mode_hook.py

```diff
@@ -0,0 +1,45 @@
+import os
+from contextlib import contextmanager
+from ray.experimental.client import ray
+
+client_mode_enabled = os.environ.get("RAY_CLIENT_MODE", "0") == "1"
```

Comment:
Is the reason for an environment variable to handle remote function definitions that occur before `ray.connect()`?

We should be able to handle this in the future so you don't need to set an env var.

Reply:
Yep, that's it. We need a way to make sure that on startup all the calls are hooked.

I'm not sure there's a great future option -- we'd have to convert or genericize the remote functions and actors, which requires some design.

For a MVP feature, a flag works surprisingly well.